include/CXXGraph/Graph/Algorithm/Dinic_impl.hpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <queue>
+#include <limits>
+#include <unordered_map>
+#include <vector>
+#include "Node_impl.h"
+#include "CXXGraph/Graph/Graph_decl.h"
+
+
+namespace CXXGraph {
+
+template <typename T>
+class Graph {
+private:
+    std::unordered_map<CXXGraph::id_t, Node<T>> nodes;
+    std::unordered_map<CXXGraph::id_t, std::unordered_map<CXXGraph::id_t, int>> capacities;
+    std::unordered_map<CXXGraph::id_t, std::unordered_map<CXXGraph::id_t, int>> flow;
```

Comment:
Note that this adds more storage to the graph, which makes it a larger object/heavier. Thoughts on this @sbaldu, @ZigRazor? Perhaps the relevant storage of the algorithm should be extracted elsewhere (and possibly the algorithm implementation itself) so that this storage is temporary.

There _are_ other options, such as lazy storage creation on the heap the first time a dfs, bfs, or Dinic's is called. Although the storage for `std::unordered_map` isn't wild - I do like keeping `Graph` as lightweight as can possibly be done.

Reply:
I Agree!